packages/flutter/lib/src/painting/image_provider.dart

```diff
@@ -764,7 +765,9 @@ class ResizeImage extends ImageProvider<_SizeAwareCacheKey> {
       );
       return decode(bytes, cacheWidth: width, cacheHeight: height, allowUpscaling: this.allowUpscaling);
     };
-    return imageProvider.load(key.providerCacheKey, decodeResize);
+    final ImageStreamCompleter completer = imageProvider.load(key.providerCacheKey, decodeResize);
+    completer.debugLabel = '${completer.debugLabel} - Resized(${key.width}×${key.height})';
```

Comment:
debugLabel should only be provided in debug build if it is using the debug naming convention.

Reply:
We have this issue in various places where things named debug have to be available in profile mode. I can guard this with a `if (!kReleaseMode)` - is that what you're getting at?

I'm not sure what a better name for this would be.

Reply:
yes `if (kReleaseMode)` would be good. Can you point me at an example where we have other `debug` fields that are filled in profile mode? The ones I was aware of were all debug build only so I'm curious which ones have made it into profile builds.

Reply:
done